lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -463,7 +494,7 @@ def expected_stanza(conditions = nil)
         expected = Hash.new { |h,k| h[k] = {} }
         return expected unless conditions
 
-        conditions[:unless_exists].try(:each) do |col|
+        conditions.delete(:unless_exists).try(:each) do |col|
```

Comment:
Yes, and that it is not is actually probably causing half those spec failures...  Thanks for noticing!